apt_info.py

```diff
@@ -80,13 +84,38 @@ def _write_held_upgrades(registry, cache):
             g.labels(change.labels['origin'], change.labels['arch']).set(change.count)
 
 
+def _write_obsolete_packages(registry, cache):
+    # This corresponds to the apt filter "?obsolete"
+    obsoletes = [p for p in cache if p.is_installed and (
+                  p.candidate is None or
+                  not p.candidate.origins or
+                  (len(p.candidate.origins) == 1 and
+                   p.candidate.origins[0].origin in ['', "/var/lib/dpkg/status"])
+                )]
+
+    g = Gauge('apt_obsolete_packages', "Apt packages which are obsolete",
+              registry=registry)
+    g.set(len(obsoletes))
+
+
 def _write_autoremove_pending(registry, cache):
     autoremovable_packages = {p for p in cache if p.is_auto_removable}
     g = Gauge('apt_autoremove_pending', "Apt packages pending autoremoval.",
               registry=registry)
     g.set(len(autoremovable_packages))
 
 
+def _write_installed_packages_per_origin(registry, cache):
+    installed_packages = {p.candidate for p in cache if p.is_installed}
+    per_origin = _convert_candidates_to_upgrade_infos(installed_packages)
+
+    if per_origin:
+        g = Gauge('apt_installed_per_origin', "Numberof ppt packages installed per origin.",
```

Comment:
similar comment here: either make this coherent with the obsolete metric, or make the obsolete metric coherent with this. :)

i would call it `apt_packages_per_origin_count`... heck, you could even call it `apt_packages_count`, then we could add more labels (e.g. `state=installed`) as we go along converting the other metrics... ;) but maybe that would be too ambiguous.

also, the description has a couple of typos here, maybe you mean:

```suggestion
        g = Gauge('apt_installed_per_origin', "Number of packages installed per origin.",
```

Reply:
idk about going with `apt_packages_count` since some packages will be double+ counted since they can match more than one state. personally because of the name I would kind of expect the total sum for `apt_packages_count` across all label variants to be the total installed packages, so if we have duplication it won't reflect that correctly.

let's go with `apt_packages_per_origin_count`

Reply:
okay, then the metrics name is definitely better..  i was actually assuming we did *not* have duplicates and could use that metric to extract a total package count, good to know that isn't the case.